apis/iam/v1beta1/diff_test.go

```diff
@@ -0,0 +1,856 @@
+						Members: []Member{"user:viewer@example.com", "user:other@example.com"},
+					},
+				},
+			},
+			actual: &IAMPolicySpec{
+				Bindings: []IAMPolicyBinding{
+					{
+						Role:    "roles/viewer",
+						Members: []Member{"user:viewer@example.com"},
+					},
+				},
+			},
+			want: &structuredreporting.Diff{
+				Fields: []structuredreporting.DiffField{
+					{
+						ID: "spec.bindings[role=roles/viewer]",
+						Old: IAMPolicyBinding{
+							Role:    "roles/viewer",
+							Members: []Member{"user:other@example.com", "user:viewer@example.com"},
+						},
+						New: IAMPolicyBinding{
+							Role:    "roles/viewer",
+							Members: []Member{"user:viewer@example.com"},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "binding members order differs - no diff",
+			desired: &IAMPolicySpec{
+				Bindings: []IAMPolicyBinding{
+					{
+						Role:    "roles/viewer",
+						Members: []Member{"user:viewer@example.com", "user:other@example.com"},
+					},
+				},
+			},
+			actual: &IAMPolicySpec{
+				Bindings: []IAMPolicyBinding{
+					{
+						Role:    "roles/viewer",
+						Members: []Member{"user:other@example.com", "user:viewer@example.com"},
+					},
+				},
+			},
+			want: &structuredreporting.Diff{},
+		},
+		{
+			name: "binding condition differs",
+			desired: &IAMPolicySpec{
+				Bindings: []IAMPolicyBinding{
+					{
+						Role:    "roles/viewer",
+						Members: []Member{"user:viewer@example.com"},
+						Condition: &IAMCondition{
+							Title:      "title",
+							Expression: "expression",
+						},
+					},
+				},
+			},
+			actual: &IAMPolicySpec{
+				Bindings: []IAMPolicyBinding{
+					{
+						Role:    "roles/viewer",
+						Members: []Member{"user:viewer@example.com"},
+						Condition: &IAMCondition{
+							Title:      "other-title",
+							Expression: "expression",
+						},
+					},
+				},
+			},
+			want: &structuredreporting.Diff{
+				Fields: []structuredreporting.DiffField{
+					{
+						ID: "spec.bindings[role=roles/viewer]",
+						Old: IAMPolicyBinding{
+							Role:    "roles/viewer",
+							Members: []Member{"user:viewer@example.com"},
+							Condition: &IAMCondition{
+								Title:      "title",
+								Expression: "expression",
+							},
+						},
+						New: IAMPolicyBinding{
+							Role:    "roles/viewer",
+							Members: []Member{"user:viewer@example.com"},
+							Condition: &IAMCondition{
+								Title:      "other-title",
+								Expression: "expression",
+							},
+						},
+					},
+				},
+			}},
+		{
+			name: "empty desired bindings",
+			desired: &IAMPolicySpec{
+				Bindings: []IAMPolicyBinding{},
+			},
+			actual: &IAMPolicySpec{
+				Bindings: []IAMPolicyBinding{
+					{
+						Role:    "roles/viewer",
+						Members: []Member{"user:viewer@example.com"},
+					},
+				},
+			},
+			want: &structuredreporting.Diff{
+				Fields: []structuredreporting.DiffField{
+					{
+						ID:  "spec.bindings[role=roles/viewer]",
+						Old: "absent in desired spec",
```

Comment:
The test `binding members order differs - no diff` is great. It would be valuable to also add a test case where one of the bindings has an empty or nil `Members` slice to ensure that it is handled correctly and doesn't cause a panic.